src/scenic/core/regions.py

```diff
@@ -281,883 +306,2816 @@ def __hash__(self):
+
+    Args:
+        visibleDistance: The view distance for this region.
+        viewAngles: The view angles for this region.
+        name: An optional name to help with debugging.
+        position: An optional position, which determines where the center of the region will be.
+        rotation: An optional Orientation object which determines the rotation of the object in space.
+        orientation: An optional vector field describing the preferred orientation at every point in
+          the region.
+        tolerance: Tolerance for collision computations.
+    """
+    def __init__(self, visibleDistance, viewAngles, name=None, position=Vector(0,0,0), rotation=None,\
+        orientation=None, tolerance=1e-8):
+        # Bound viewAngles from either side.
+        viewAngles = tuple([min(angle, math.tau) for angle in viewAngles])
+
+        if min(viewAngles) <= 0:
+            raise ValueError("viewAngles cannot have a component less than or equal to 0")
+
+        # Cases in view region computation
+        # Case 1:       Azimuth view angle = 360 degrees
+        #   Case 1.a:   Altitude view angle = 180 degrees   => Full Sphere View Region
+        #   Case 1.b:   Altitude view angle < 180 degrees   => Sphere - (Cone + Cone) (Cones on z axis expanding from origin)
+        # Case 2:       Azimuth view angle = 180 degrees
+        #   Case 2.a:   Altitude view angle = 180 degrees   => Hemisphere View Region
+        #   Case 2.b:   Altitude view angle < 180 degrees   => Hemisphere - (Cone + Cone) (Cones on appropriate hemispheres)
+        # Case 3:       Altitude view angle = 180 degrees   
+        #   Case 3.a:   Azimuth view angle < 180 degrees    => Sphere intersected with Pyramid View Region
+        #   Case 3.b:   Azimuth view angle > 180 degrees    => Sphere - Backwards Pyramid View Region 
+        # Case 4:       Both view angles < 180              => Capped Pyramid View Region
+        # Case 5:       Azimuth > 180, Altitude < 180       => (Sphere - (Cone + Cone) (Cones on appropriate hemispheres)) - Backwards Capped Pyramid View Region
+
+        view_region = None
+        diameter = 2*visibleDistance
+        base_sphere = SpheroidRegion(dimensions=(diameter, diameter, diameter), engine="scad")
+
+        if (math.tau-0.01 <= viewAngles[0] <= math.tau+0.01):
+            # Case 1
+            if viewAngles[1] > math.pi-0.01:
+                #Case 1.a
+                view_region = base_sphere
+            else:
+                # Case 1.b
+                # Create cone with yaw oriented around (0,0,-1)
+                padded_height = visibleDistance * 2
+                radius = padded_height*math.tan((math.pi-viewAngles[1])/2)
+
+                cone_mesh = trimesh.creation.cone(radius=radius, height=padded_height)
+
+                position_matrix = translation_matrix((0,0,-1*padded_height))
```

Comment:
Shouldn't this be `-padded_height/2`?

Reply:
I don't believe so, since the position might not be the center in this case.

Reply:
The documentation for `trimesh.creation.cone` says it creates the cone centered at the origin; I assumed that meant the center of the bounding box, but maybe not?

Reply:
I just checked and it looks like they mean centered on x and y, but not z. The center base of the cone is at the origin by default.